packages/lib-user/src/components/UserStats/UserStats.spec.js

```diff
@@ -0,0 +1,73 @@
+import { render, screen } from '@testing-library/react'
+import proxyquire from 'proxyquire'
+import sinon from 'sinon'
+
+import { PROJECTS, USER } from '../../../test/mocks/panoptes.mock.js'
+import { STATS } from '../../../test/mocks/stats.mock.js'
+
+describe('components > UserStats', function () {
+  this.timeout(5000)
+  
+  let mockAuthClient
+
+  let usePanoptesProjectsStub
+  let usePanoptesUserStub
+  let useUserStatsStub
+
+  let UserStats
+
+  // happy path
+  before(function () {
+    mockAuthClient = {
+      checkBearerToken: sinon.stub().resolves('fake token'),
+      checkCurrent: sinon.stub().resolves(USER),
+      getBearerToken: sinon.stub().resolves('fake token'),
+      listen: sinon.stub(),
+      stopListening: sinon.stub()
+    }
+
+    // Create stubs for the hooks
+    usePanoptesProjectsStub = sinon.stub().returns({ data: PROJECTS, error: null, isLoading: false })
+    usePanoptesUserStub = sinon.stub().returns({ data: USER, error: null, isLoading: false })
+    useUserStatsStub = sinon.stub().returns({ data: STATS, error: null, isLoading: false })
+
+    // Use proxyquire to replace the hooks with the stubs
+    UserStats = proxyquire('./UserStats', {
```

Comment:
Just copying over my Slack reply here to document ~
> I think the way you’re using proxyquire for UserStats is a helpful strategy! It looks like you’re stubbing the custom hooks because UserStats performs several fetch functions (the custom hooks) and then handles the data before passing it to child UI components. And your unit tests are for happy states when data exists and sad states when there’s an error fetching :+1:

I like it! The only thing to be wary of while using stubs and libraries like sinon, nock, and proxyquire is to avoid testing implementation like we often see in lingering enzyme tests (some look for a function to be called with a specific variable, but don’t tell us anything about the UI). UserStats’s specs look like a good start though!